protostar/commands/init/init_command.py

```diff
@@ -1,21 +1,27 @@
 import glob
-import shutil
-from pathlib import Path
-from typing import Any, List, Optional, Type
-
-from git.exc import InvalidGitRepositoryError
-from git.repo import Repo
+from typing import List, Optional
 
 from protostar.cli import Command
-from protostar.utils import VersionManager, log_color_provider
-from protostar.utils.config.project import Project, ProjectConfig
+from protostar.commands.init.project_creator.adapted_project_creator import (
+    AdaptedProjectCreator,
+)
+from protostar.commands.init.project_creator.new_project_creator import (
+    NewProjectCreator,
+)
+from protostar.utils.requester import Requester
 
 
 class InitCommand(Command):
-    def __init__(self, script_root: Path, version_manager: VersionManager) -> None:
+    def __init__(
+        self,
+        requester: Requester,
+        new_project_creator: NewProjectCreator,
```

Comment:
I think we shouldn't inject creators here. I think Init command is the one which should be responsible for creating this objects. Can we change it?

Reply:
I see the following issues with such approach:

- Creating objects in this class would require passing `ProjectCreator`'s dependencies to `InitCommand`.
- Stategy Pattern can be implemented with DI https://ufukhaciogullari.com/blog/strategy-pattern-with-dependency-injection/
- Objects should be created in the composition root https://blog.ploeh.dk/2011/07/28/CompositionRoot/
  - (I should've created `ProjectCreatorResolver`)
- DI framework will be easier to integrate if we decide to use it in the future
- Testability is better with current approach
- not following open–closed principle